core/discovery/discovery.go

```diff
@@ -95,6 +97,7 @@ func (d *Discovery) Start(ownIdentity identity.Identity, proposal market.Service
 	d.ownIdentity = ownIdentity
 	d.signer = d.signerCreate(ownIdentity)
 	d.proposal = proposal
+	d.chainID = config.GetInt64(config.FlagChainID)
```

Comment:
either inject it to the discovery, or just use it from config directly where needed without having it saved in discovery